### PR TITLE
add python cache removal to `spack clean`

### PR DESCRIPTION
**Pull request by healther**
Continuing the discussion from #8090, this is a quick way to clean up spack's python caches.

Would have probably been helpful to have this ability in #8248, #8267, #8240, #8292

### CODE REVIEW

File: lib/spack/spack/cmd/clean.py [hunk 1]
```diff
@@ -88,3 +93,17 @@ def clean(parser, args):
     if args.misc_cache:
         tty.msg('Removing cached information on repositories')
         spack.caches.misc_cache.destroy()
+
+    if args.python_cache:
+        tty.msg('Removing python cache files.')
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/clean.py]: I would leave off the period to be consistent with other messages.

File: lib/spack/spack/cmd/clean.py [hunk 2]
```diff
@@ -88,3 +93,17 @@ def clean(parser, args):
...
+        for root, dirs, files in os.walk(os.environ['SPACK_ROOT']):
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/clean.py]: This environment variable is not set by default, only if you happen to use `setup-env.sh`. You should use this instead:
```python
from spack.paths import spack_root

for root, dirs, files in os.walk(spack_root):
```

File: lib/spack/spack/cmd/clean.py [hunk 3]
```diff
@@ -88,3 +93,17 @@ def clean(parser, args):
...
+            for f in files:
+                if f.endswith('.pyc') or f.endswith('.pyo'):
+                    fname = os.path.join(root, f)
+                    tty.debug('removing {0}'.format(fname))
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/clean.py]: I would capitalize this to be consistent with other messages.

File: lib/spack/spack/cmd/clean.py [hunk 4]
```diff
@@ -88,3 +93,17 @@ def clean(parser, args):
...
+                    os.remove(fname)
+            for d in dirs:
+                if d == '__pycache__':
+                    dname = os.path.join(root, d)
+                    tty.debug('removing {0}'.format(dname))
```
Comment (adamjstewart) [inline lib/spack/spack/cmd/clean.py]: I would capitalize this to be consistent with other messages.

**Comment by scheibelp** [comment]
Thanks @healther for the patch!

Thanks @adamjstewart for reviewing!

**Pull request merged**